util/chplenv/chpl_arch.py

```diff
@@ -550,6 +549,23 @@ def get(location, map_to_compiler=False, get_lcd=False):
     return arch_tuple(flag or 'none', arch or 'unknown')
 
 
+# Returns the default machine.
+#  * if arch is native/none/unknown, return result of get_native_machine
+#  * otherwise returns the machine type corresponding to the selected arch
+@memoize
+def get_default_machine(location):
+    (flag, arch) = get(location)
+
+    # Native/none/unknown just get the native machine by default
+    if arch == "native" or arch == "none" or arch == "unknown":
+      return get_native_machine()
+    # Otherwise, compute the machine based on the architecture selected
+    if is_known_arm(arch):
```

Comment:
This seems likely to easily become out of date.

Reply:
I agree. I think we could do something better using what we already have in `feature_sets` but I thought that'd be a task for some one with more Python skills than me. I view this version as an OK 1st solution, though.